plugins/modules/foreman_compute_resource.py

```diff
@@ -112,133 +150,91 @@
-            locations=dict(type='list'),
-            organizations=dict(type='list'),
+            provider_params=dict(type='dict', options=dict(
+                url=dict(),
+                display_type=dict(),
+                user=dict(),
+                password=dict(no_log=True),
+                datacenter=dict(),
+                use_v4=dict(type='bool'),
+                ovirt_quota=dict(),
+            )),
             state=dict(type='str', default='present', choices=['present', 'absent', 'present_with_defaults']),
-
-            # Deprecated provider's specific params, use nested keys in provider_params param instead
-            provider_auth=dict(type='dict'),
-            url=dict(type='str'),
-            display_type=dict(type='str'),
-            datacenter=dict(type='str'),
         ),
         required_if=(
             ['state', 'present', ['provider']],
         ),
     )
 
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     if 'provider' in entity_dict:
         entity_dict['provider'] = entity_dict['provider'].title()
 
     provider_infos = get_provider_infos(provider=entity_dict.get('provider', ''))
     provider_params = entity_dict.pop('provider_params', dict())
-    provider_auth = entity_dict.pop('provider_auth', dict())
 
     module.connect()
 
-    try:
-        # Try to find the compute_resource to work on
-        entity = find_compute_resource(module, name=entity_dict['name'], failsafe=True)
-    except Exception as e:
-        module.fail_json(msg='Failed to find entity: %s ' % e)
+    entity = module.find_resource_by_name('compute_resources', name=entity_dict['name'], failsafe=True)
 
-    if 'updated_name' in entity_dict and state == 'present':
-        entity_dict['name'] = entity_dict['updated_name']
+    if not module.desired_absent:
+        if 'updated_name' in entity_dict:
+            entity_dict['name'] = entity_dict['updated_name']
 
-    if 'organizations' in entity_dict:
-        entity_dict['organizations'] = find_organizations(module, entity_dict['organizations'])
+        if 'organizations' in entity_dict:
+            entity_dict['organizations'] = module.find_resources_by_name('organizations', entity_dict['organizations'], thin=True)
 
-    if 'locations' in entity_dict:
-        entity_dict['locations'] = find_locations(module, entity_dict['locations'])
+        if 'locations' in entity_dict:
+            entity_dict['locations'] = module.find_resources_by_title('locations', entity_dict['locations'], thin=True)
 
-    entity_dict = sanitize_entity_dict(entity_dict, name_map)
-
-    # Add provider specific params
-    if state in ['present', 'present_with_defaults']:
+        # Add provider specific params
         if not provider_infos and not entity:
             module.fail_json(msg='To create a compute resource a valid provider must be supplied')
 
         for key in provider_infos['params']:
```

Comment:
should we add a check that `provider_params` has no entries that are not in `provider_infos['params']`?

smth like:
```python
given_provider_params = set(provider_params.keys())
supported_provider_params = set(provider_infos['params'])
unsupported_provider_params = given_provider_params - supported_provider_params
if unsupported_provider_params:
  module.fail_json(msg="Provider {} does not support the following params: {}".format(entity_dict.get('provider'), unsupported_provider_params))
```

Reply:
Yes, i like this.